File: backend/app/models/task_log.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.core.database import Base


class TaskLog(Base):
    __tablename__ = "task_logs"

    id = Column(Integer, primary_key=True, index=True)
    task_id = Column(Integer, ForeignKey("tasks.id"), nullable=False)
    content = Column(Text, nullable=False)  # Markdown 内容
    log_type = Column(String(20), default="note")  # note/problem/knowledge/progress
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    task = relationship("Task", backref="logs")

File: frontend/src/api/taskLog.ts
import api from './index'

export interface TaskLog {
  id: number
  task_id: number
  content: string
  log_type: string
  created_at: string
  updated_at: string
}

export const taskLogApi = {
  list: (params?: { task_id?: number; log_type?: string; limit?: number }) =>
    api.get<any, TaskLog[]>('/api/task-logs', { params }),
  get: (id: number) => api.get<any, TaskLog>(`/api/task-logs/${id}`),
  create: (data: Partial<TaskLog>) => api.post('/api/task-logs', data),
  update: (id: number, data: Partial<TaskLog>) => api.put(`/api/task-logs/${id}`, data),
  delete: (id: number) => api.delete(`/api/task-logs/${id}`),
  recent: (limit?: number) => api.get<any, TaskLog[]>('/api/task-logs/recent', { params: { limit } }),
}
